docs/Optimize-KernelBuild.md

```diff
@@ -0,0 +1,85 @@
+# How to optimize the Linux kernel build
+
+The kernel build script which this tutorial addresses is found here:
+https://github.com/LIS/lis-pipeline/blob/master/scripts/package_building/build_artifacts.sh
+
+
+## Environment minimum requirements
+  - Build OSes: Ubuntu Xenial 16.04 or CentOS/RHEL 7.
+  - CPU: minimum 16 cores, more the better.
+  - RAM: minimum 4GB, as one kernel build does not use more than 1 or 2 GB of RAM.
+  - IO: SSD required, minimum of 5k IOPS. On an Azure VM you can add 12 disks of premium storage and configure them into RAID 0 and you can achieve 60k IOPS. More the better.
+
+## Required packages
+### Debian
+  - Packages required:
+  ```bash
+    set -xe
+    deb_packages=(libncurses5-dev xz-utils libssl-dev libelf-dev bc ccache kernel-package \
+        devscripts build-essential lintian debhelper git wget bc fakeroot flex bison asciidoc \
+        pigz pbzip2)
+    DEBIAN_FRONTEND=noninteractive sudo apt-get -y install ${deb_packages[@]}
+  ```
+
+### RHEL
+  - Packages required:
+  ```bash
+    set -xe
+    rpm_packages=(rpm-build rpmdevtools yum-utils ncurses-devel hmaccalc zlib-devel \ 
+        binutils-devel elfutils-libelf-devel openssl-devel wget git ccache bc fakeroot \
+        asciidoc audit-devel binutils-devel xmlto bison flex gtk2-devel libdw-devel libelf-devel \
+        xz-devel libnuma-devel newt-devel openssl-devel xmlto zlib-devel pigz pbzip2)
+    sudo yum groups mark install "Development Tools"
+    sudo yum -y groupinstall "Development Tools"
+    sudo yum -y install ${rpm_packages[@]}
+  ```
+
+## Optimizations
+### Generic optimizations
+  - Configure the number of cores for the make command to be times 2 or 3 the number of physical cores.
+    This will speed up the build process and the only bottleneck will be the IOPS.
+  - Use ccache: https://ccache.samba.org/
```

Comment:
Add the bash script 
sudo apt-get update && sudo apt-get install ccache pigz pbzip2 -y